sway_libs/src/string/SPECIFICATION.md

```diff
@@ -0,0 +1,63 @@
+# Overview
+
+This document provides an overview of the String library.
+
+It outlines the use cases, i.e. specification, and describes how to implement the library.
+
+## Use Cases
+
+The String library can be used anytime a string's length is unknown at compile time and as such lives on the heap. Further methods can then be implemented to provide additional features building off of the `String` struct.

```

Comment:
```suggestion
The String library can be used anytime a string's length is unknown at compile time. Further methods can then be implemented to provide additional features building off of the `String` struct.
```

Irrelevant for a use case description. The user only cares about having a dynamic string rather than the static built-in. The implementation can be mentioned in some technical section somewhere if it's relevant